source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -303,6 +303,31 @@ const std::string& SslSocket::urlEncodedPemEncodedPeerCertificate() const {
   return cached_url_encoded_pem_encoded_peer_certificate_;
 }
 
+const std::string& SslSocket::urlEncodedPemEncodedPeerCertificateChain() const {
+  if (!cached_url_encoded_pem_encoded_peer_cert_chain_.empty()) {
+    return cached_url_encoded_pem_encoded_peer_cert_chain_;
+  }
+
+  STACK_OF(X509)* cert_chain = SSL_get_peer_cert_chain(ssl_.get());
```

Comment:
Note: you might consider using `SSL_get_peer_full_cert_chain()` instead of `SSL_get_peer_cert_chain()` if you want to always include the leaf certificate in the chain.

It seems that the tool referenced in #6135 might want only the intermediate certificates, but looking at the tests in this PR, I believe that the assumption is that the leaf certificate is always included.

Per BoringSSL's documentation and code, the leaf certificate isn't included in `SSL_get_peer_cert_chain()` when SSL object is a server (but it looks like it's returned in the tests, which is a bit confusing...).

Reply:
`SSL_get_peer_cert_chain(..)` returns only the intermediate cert in the test. Isn't that what's expected?

Reply:
Switched to `SSL_get_peer_full_cert_chain(..)` in f4c974d.

Reply:
It does, indeed.

To be honest, returning only intermediate certs is not what I expected, since "certificate chain" in my mind includes the leaf certificate, but that doesn't mean it should include it.

Let's leave it as-is (unless other people object), but could you make it more explicit in the comments, e.g.:
```
   * @return std::string the URL-encoded PEM-encoded representation of the peer certificate chain
   *         (intermediate certificates only, no leaf certificate).
   *         Returns "" if there is no peer certificate or encoding fails.
```
Thanks!

Reply:
I did it the other way. I'm including the leaf cert in the chain and have updated the documentation accordingly.

Reply:
Sounds good, thanks!

@lizan any objections to returning full certificate chain?

Reply:
nope, that SGTM